File: 561. Array Partition.java
// Time Complexity: O(N LOG N)
// Space Complexity: O(1)
// Did this code successfully run on Leetcode: YES
// Any problem you faced while coding this: NO

class Solution {
    public int arrayPairSum(int[] nums) {
        if(nums == null || nums.length == 0) return 0;
        
        int sum = 0;
        Arrays.sort(nums);
        for(int i = 0; i < nums.length; i = i + 2){
            sum += nums[i];
        }
        
        return sum;
    }
}S

File: 561. Array Partition.cpp
//Time: O(n)
//Space: O(n)

//Greedy: Sort, pick minimum two elements each time so that the effect of minimization (we need to take minimum value of that pair) is reduced
class Solution {
public:
    int arrayPairSum(vector<int>& v) {
        sort(v.begin(), v.end());

        int sum = 0;
        for(int i=0;i<v.size()-1;i+=2){
            sum += min(v[i],v[i+1]);
        }
        return sum;
    }
};
